trulens_eval/trulens_eval/tru.py

```diff
@@ -745,6 +746,42 @@ def get_leaderboard(
 
         return leaderboard
 
+    def get_leaderboard_grouped_by_metadata(
+        record_metadata_key: str = None,
```

Comment:
how about we just allow an extra optional field sth like `sort_by_record_metadata_key` to the existing  `get_leaderboard` method above rather than adding a new method? 

`get_leaderboard_grouped_by_metadata` looks similar enough to me to the original `get_leaderboard()`

Reply:
Good point - done

Reply:
Good thinking - made this update!